tmt/steps/report/reportportal.py

```diff
@@ -0,0 +1,89 @@
+import dataclasses
+import io
+import os
+import zipfile
+from typing import List, Optional
+
+import click
+import junit_xml
+
+import tmt.steps.report
+import tmt.utils
+
+from . import junit
+
+
+@dataclasses.dataclass
+class ReportPortalData(tmt.steps.report.ReportStepData):
+    url: Optional[str] = None
+    token: Optional[str] = None
+    launch_name: Optional[str] = None
+
+
+@tmt.steps.provides_method("reportportal")
+class ReportPortal(tmt.steps.report.ReportPlugin):
```

Comment:
Thinking a bit about the name. Currently we have:

    ReportDisplay
    ReportHtml
    ReportJUnit
    ReportPolarion

So in order to make the naming consistent we should probably call the new class `ReportReportPortal` so that the suffix covers the whole `ReportPortal` name.

Reply:
Or we should go the other way around and call the plugin `Portal` only? The command line would be a bit shorter ;-)

    tmt run report --how portal ...

Would probably be misleading. Thoughts?

Reply:
`ReportReportPortal` may be ugly in the source code but it is more correct.
`... --how reportportal` seems correct to me.
Perhaps, because the upstream name is *ReportPortal*, there is implied separation between words, we should use our own preferren separator, therefore having `... --how report_portal` or `... --how report-portal`.

Reply:
Ok, let's keep the naming consistent with upstream: They use `ReportPortal` without any space, so let's have the `ReportReportPortal` class and lower case `reportportal` when selecting the plugin from the command line.